library/src/test/java/com/datatorrent/lib/io/fs/FileSplitterInputTest.java

```diff
@@ -499,7 +499,7 @@ public void testRecoveryOfBlockMetadataIterator() throws InterruptedException
     testMeta.fileSplitterInput.getScanner().setScanIntervalMillis(500);
 
 
-    testMeta.fileSplitterInput.setup(testMeta.context);
```

Comment:
For the first point, we can (should) add another unit test just to test FileSplitterInput.setup() which in any case sounds like a better idea.

The 2nd point is not very clear. Are you saying there are many other instances of non-transient fields which should be checked for null in setup() or are you saying there are many fields which should ideally be declared transient?